cmd/amplifier/server/server.go

```diff
@@ -53,28 +65,32 @@ var serviceInitializers = []serviceInitializer{
 	registerClusterServer,
 }
 
-// Start starts the amplifier server
-func Start(c *Configuration) {
-	// initialize clients
-	initClients(c)
+func New(config *configuration.Configuration) *Amplifier {
+	amp := &Amplifier{config: config}
+	initClients(amp)
+	amp.accounts = accounts.NewStore(amp.store, amp.config)
+	return amp
+}
 
+// Start starts the amplifier server
+func (a *Amplifier) Start() {
 	// register services
 	s := grpc.NewServer(
 		grpc.StreamInterceptor(auth.StreamInterceptor),
 		grpc.UnaryInterceptor(auth.Interceptor),
 	)
-	registerServices(c, s)
+	registerServices(a, s)
 
 	// start listening
-	lis, err := net.Listen("tcp", c.Port)
+	lis, err := net.Listen("tcp", a.config.Port)
 	if err != nil {
-		log.Fatalf("Unable to listen on %s: %v\n", c.Port[1:], err)
+		log.Fatalf("Unable to listen on %s: %v\n", a.config.Port[1:], err)
 	}
-	log.Println("Listening on port:", c.Port[1:])
+	log.Println("Listening on port:", a.config.Port[1:])
 	log.Fatalln(s.Serve(lis))
 }
 
-func initClients(config *Configuration) {
+func initClients(amp *Amplifier) {
```

Comment:
The changes look good. My only observation is that instead of passing around all these pointers, perhaps we should just make all of these Amplifier methods.